examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1468 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the deal.II distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Daniel Garcia-Sanchez, CNRS, 2019
+ */
+
+// @sect3{Include files}
+
+// Most of the include files we need for this program have already been
+// discussed in previous programs, in particular in step-40.
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/function.h>
+
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/generic_linear_algebra.h>
+#include <deal.II/lac/petsc_solver.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/error_estimator.h>
+
+#include <fstream>
+#include <iostream>
+
+// The following header provides the Tensor class that we use represent the
+// material properties.
+#include <deal.II/base/tensor.h>
+
+
+// The following header is necessary for the HDF5 interface of deal.II.
+#include <deal.II/base/hdf5.h>
+
+// This header is required for the function VectorTools::point_value that we use
+// to read the result of the simulation.
+
+#include <deal.II/numerics/vector_tools.h>
+
+// We need this header for the function GridTools::find_active_cell_around_point
+// that we use in the function ElasticWave<dim>::store_frequency_step_data
+#include <deal.II/grid/grid_tools.h>
+
+namespace step62
+{
+  using namespace dealii;
+
+  // @sect3{Auxiliary classes and functions}
+  // The following classes are used to store the parameters of the simulation.
+
+  // @sect4{`RightHandSide` class}
+  // This class is used to define the force pulse on the left side of the
+  // structure.
+  template <int dim>
+  class RightHandSide : public Function<dim>
+  {
+  public:
+    RightHandSide(HDF5::Group &data);
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component) const;
+
+  private:
+    // `data` is the HDF5::Group in which all the simulation results will be
+    // stored. Note that this variable points to the same HDF5::Group of
+    // `RightHandSide::data`, `PML::data` and `Parameters::data`. When a
+    // HDF5::Group is copied, it will point to the same HDF5 Group; this is
+    // achieved with the protected std::shared_ptr<hid_t>
+    // HDF5::Group::hdf5_reference.
```

Comment:
I think you are referring to internals of the `HDF5::Group` class here, but from a user perspective this is not of interest. Can you rephrase this in a way that explains *what* is happening, not *why*?